token/js/src/extensions/immutableOwner.ts

```diff
@@ -0,0 +1,24 @@
+import { struct } from '@solana/buffer-layout';
+import { publicKey } from '@solana/buffer-layout-utils';
+import { PublicKey } from '@solana/web3.js';
+import { Mint } from '../state/mint';
+import { ExtensionType, getExtensionData } from './extensionType';
+
+/** MintCloseAuthority as stored by the program */
+export interface ImmutableOwner {
+    owner: PublicKey;
```

Comment:
This is tricky -- there is no data in the immutable owner extension, so you really only need to check its existence